pandas/tests/io/test_sql.py

```diff
@@ -538,13 +536,12 @@ class DummyException(Exception):
 
         # Make sure when transaction is rolled back, no rows get inserted
         ins_sql = "INSERT INTO test_trans (A,B) VALUES (1, 'blah')"
-        try:
+
+        with pytest.raises(DummyException):
```

Comment:
I don't think we need to assert the exception here, from reading the comment we are just raising it to ensure the transaction is correctly rolled back. I think using a try/except with ignoring the exception makes that clearer.